src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1408,99 +1409,185 @@ static constexpr bool _isHexNumber(const wchar_t wch) noexcept
+            // 9 "rgb:h/h/h"
+            // 12 "rgb:hh/hh/hh"
+            // 15 "rgb:hhh/hhh/hhh"
+            // 18 "rgb:hhhh/hhhh/hhhh"
+            // Any fewer cannot be valid, and any more will be too many.
+            // Return early in this case.
+            if (stringSize < 9 || stringSize > 18)
+            {
+                return false;
+            }
+
+            foundRGBColorSpec = true;
+
+            std::advance(curr, 4);
+        }
     }
 
-    // Now we look for "rgb:"
-    // Other colorspaces are theoretically possible, but we don't support them.
+    // Try the sharp sign format.
+    if (!foundRGBColorSpec && stringSize > 1)
+    {
+        const auto prefix = string.substr(0, 1);
+        if (prefix.compare(L"#") == 0)
+        {
+            // We can have one of the following formats:
+            // 4 "#hhh"
+            // 7 "#hhhhhh"
+            // 10 "#hhhhhhhhh"
+            // 13 "#hhhhhhhhhhhh"
+            // Any other cases will be invalid.
+            // Return early in this case.
+            if (!(stringSize == 4 || stringSize == 7 || stringSize == 10 || stringSize == 13))
+            {
+                return false;
+            }
 
-    auto curr = string.cbegin();
-    if ((*curr++ == L'r') &&
-        (*curr++ == L'g') &&
-        (*curr++ == L'b') &&
-        (*curr++ == L':'))
+            isSharpSignFormat = true;
+            foundRGBColorSpec = true;
+            rgbHexDigitCount = (stringSize - 1) / 3;
+
+            std::advance(curr, 1);
+        }
+    }
+
+    // Try the XOrg app color name.
+    if (!foundRGBColorSpec)
     {
-        foundRGB = true;
+        try
+        {
+            til::color color = Utils::ColorFromXOrgAppColorName(string);
+            rgb = RGB(color.r, color.g, color.b);
+            success = true;
+        }
+        catch (...)
+        {
+        }
     }
 
-    if (foundRGB)
+    if (foundRGBColorSpec)
     {
-        // Colorspecs are up to hh/hh/hh, for 1-2 h's
         for (size_t component = 0; component < 3; component++)
         {
             bool foundColor = false;
-            auto& value = til::at(colorValues, component);
-            for (size_t i = 0; i < 3; i++)
+            auto& parameterValue = til::at(parameterValues, component);
+            // For "sharp sign" format, the rgbHexDigitCount is known.
+            // For "rgb:" format, colorspecs are up to hhhh/hhhh/hhhh, for 1-4 h's
+            const auto iteration = isSharpSignFormat ? rgbHexDigitCount : 4;
+            for (size_t i = 0; i < iteration && curr < string.cend(); i++)
             {
                 const wchar_t wch = *curr++;
 
-                if (_isHexNumber(wch))
+                if (!_isHexNumber(wch))
+                {
+                    return false;
+                }
+
+                parameterValue *= 16;
+                unsigned int intVal = 0;
+                const auto ret = s_HexToUint(wch, intVal);

```

Comment:
Is it just me or the `s_HexToUint` feels the same as `_isHexNumber` above?

Reply:
Yeah you're right. I don't think you need the `_isHexNumber` check.

Reply:
And this is actually the only usage of `_isHexNumber` function...